dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -548,6 +548,15 @@ private void doExportUrls(RegisterTypeEnum registerType) {
         final boolean serverService = ref instanceof ServerService;
         if (serverService) {
             serviceDescriptor = ((ServerService) ref).getServiceDescriptor();
+            ProviderConfig providerConfig = getScopeModel()
```

Comment:
Is it possible to do this in a simpler way?

Reply:
Has the `url` instance already been generated at this point?